### PR TITLE
Add dynamic CSS and JS support for gluegun

### PR DESCRIPTION
**Pull request by deekoder**

### CODE REVIEW

File: lib/_sidebar.erb [hunk 1]
```diff
@@ -13,14 +13,28 @@
...
       <!-- Tab content -->
       <div class="tab-content">
-        <div class="tab-pane active text-style" id="content"></div>
+      <%- @site_map['docs'].each do |key, value| -%>
+        <%- first_entry = true -%>
+        <%- value.each do |key2, value2| -%>
+          <%- if key===@site_map.keys[0] && first_entry -%>
+            <%- active_keyword="active tab-pane text-shtmltyle" -%>
+            <%- first_entry = false -%>
+          <%- else -%>
```
Comment (ebozduman) [inline lib/_sidebar.erb]: 1. You don't need the `else` part of the if clause. It is just 2 lines shorter this way.

Combining this comment together with the comment above about line# 26, the code becomes:
```
            <%- value.keys.each do |key2| -%>
                <%- active_keyword="tab-pane text-style" -%>
                <%- if key===@site_map.keys[0] && first_entry -%>
                    <%- active_keyword="active tab-pane text-shtmltyle" -%>
                    <%- first_entry = false -%>
                <% - end -%>
                <div role="tabpanel" class="<%=active_keyword%>" id="<%=key2["slug"]%>">
                    <%- linked_doc = key2["link"] -%>
                    <%=  reveal(linked_doc) -%>
                </div>
            <%- end -%>
```

File: lib/gluegun.rb [hunk 2]
```diff
@@ -17,6 +17,7 @@ def self.gluegun_generate_index(site_config_file)
       @site_map = YAML.load(open(config_file).read)
       dest_path = @site_map['destination']
       FileUtils.mkdir_p(dest_path) unless File.exist?(dest_path)
+      #FileUtils.cp_r(Dir.glob('css/*'), dest_path, :verbose =>true)
```
Comment (ebozduman) [inline lib/gluegun.rb]: You can remove the comment.

**Comment by ebozduman** [comment]
Sorry for mistakenly clicking on `Approve` radio button, while comments asking some questions/suggestions.

**Comment by deekoder** [comment]
i'll make these changes and resend

File: lib/_sidebar.erb [hunk 3]
```diff
@@ -13,14 +13,28 @@
...
             <%- end -%>
           </ul>
         </nav>
       </div>
       <!-- Tab content -->
       <div class="tab-content">
-        <div class="tab-pane active text-style" id="content"></div>
+      <%- @site_map['docs'].each do |key, value| -%>
+        <%- first_entry = true -%>
+        <%- value.each do |key2, value2| -%>
```
[the remark this answers is not in the document]
Comment (deekoder) [reply lib/_sidebar.erb]: i was printing this and hence used this style. I will refactor here.

File: lib/_sidebar.erb [hunk 4]
```diff
@@ -13,14 +13,28 @@
...
+          <%- if key===@site_map.keys[0] && first_entry -%>
+            <%- active_keyword="active tab-pane text-shtmltyle" -%>
```
[the remark this answers is not in the document]
Comment (deekoder) [reply lib/_sidebar.erb]: this is a typo. will fix.

File: lib/_sidebar.erb [hunk 5]
```diff
@@ -13,14 +13,28 @@
...
+                <li class="<%=active_keyword%>"><a href="#<%=key2["slug"]%>" data-toggle="tab"><%=key2["title"]%></a></li>
               <%- end -%>
             <%- end -%>
           </ul>
         </nav>
       </div>
       <!-- Tab content -->
       <div class="tab-content">
-        <div class="tab-pane active text-style" id="content"></div>
+      <%- @site_map['docs'].each do |key, value| -%>
```
[the remark this answers is not in the document]
Comment (deekoder) [reply lib/_sidebar.erb]: Ersan key, value is std terminology so i am inclined to not change this. Let me know if you feel strongly about it.

[the remark this answers is not in the document]
Comment (ebozduman) [reply lib/_sidebar.erb]: That's fine.

**Pull request closed without merging**